test/query_test.exs

```diff
@@ -8,6 +8,69 @@ defmodule QueryTest do
     {:ok, [pid: pid]}
   end
 
+  test "support primitive data types", context do
+    integer          = 1
+    negative_integer = -1
+    float            = 3.1415
+    negative_float   = -3.1415
+    string           = "Californication"
+    text             = "Some random text"
+    binary           = <<0,1>>
+    table            = "basic_types"
+
+    sql = ~s{CREATE TABLE #{table} } <>
+          ~s{(id serial, active boolean, count integer, intensity float, } <>
+          ~s{title varchar(20), body text(20), data blob)}
+
+    :ok = query(sql, [])
+
+    # Booleans
+    :ok = query("INSERT INTO #{table} (active) values (?)", [true])
+    [{true}] = query("SELECT active from #{table} WHERE id = LAST_INSERT_ID()", [])
+
+    # Integer
+    :ok = query("INSERT INTO #{table} (count) values (?)", [integer])
+    [{^integer}] = query("SELECT count from #{table} WHERE id = LAST_INSERT_ID()", [])
+    :ok = query("INSERT INTO #{table} (count) values (?)", [negative_integer])
+    [{^negative_integer}] = query("SELECT count from #{table} WHERE id = LAST_INSERT_ID()", [])
+
+    # Float
+    :ok = query("INSERT INTO #{table} (intensity) values (?)", [float])
+    [{^float}] = query("SELECT intensity from #{table} WHERE id = LAST_INSERT_ID()", [])
+    :ok = query("INSERT INTO #{table} (intensity) values (?)", [negative_float])
+    [{^negative_float}] = query("SELECT intensity from #{table} WHERE id = LAST_INSERT_ID()", [])
+
+    # String
+    :ok = query("INSERT INTO #{table} (title) values (?)", [string])
+    [{^string}] = query("SELECT title from #{table} WHERE id = LAST_INSERT_ID()", [])
+    [{"mø"}] = query("SELECT 'mø'", [])
+
+    # Text
+    :ok = query("INSERT INTO #{table} (body) values (?)", [text])
+    [{^text}] = query("SELECT body from #{table} WHERE id = LAST_INSERT_ID()", [])
+
+    # Binary
+    :ok = query("INSERT INTO #{table} (data) values (?)", [binary])
+    [{^binary}] = query("SELECT data from #{table} WHERE id = LAST_INSERT_ID()", [])
+
+    # Nil
+    [{nil}] = query("SELECT null", [])
+  end
+
+  test "encode and decode decimals", context do
+    table            = "test_decimals"
+    :ok = query("CREATE TABLE #{table} (id serial, cost decimal(10,4))", [])
+
+    :ok = query("INSERT INTO #{table} (cost) values (?)", [Decimal.new("12.93")])
+    assert [{Decimal.new("12.9300")}] == query("SELECT cost FROM #{table}", [])
+
+    :ok = query("INSERT INTO #{table} (cost) values (?)", [Decimal.new("-164.9")])
+    assert [{Decimal.new("-164.9000")}] == query("SELECT cost FROM #{table} WHERE id = LAST_INSERT_ID()", [])
+
+    :ok = query("INSERT INTO #{table} (cost) values (?)", [Decimal.new("16400")])
+    assert [{Decimal.new("16400.0000")}] == query("SELECT cost FROM #{table} WHERE id = LAST_INSERT_ID()", [])
```

Comment:
That looks fine. You can also use `Decimal.equal?` if you don't want to depend on the precision the server sends.